posydon/binary_evol/simulationproperties.py

```diff
@@ -147,6 +147,11 @@ def pre_step(self, binary, step_name):
         binary : instance of <class, BinaryStar>
 
         """
+        ## do not call extra step hooks if history_verbose=False
+        if not binary.history_verbose and binary.event is not None:
+            if "redirect" in binary.event:
+                return binary
```

Comment:
Are you sure, we need this here? I have my doubts, that `binary.event` is well defined pre step?
I'd guess, it's either undefined or copied from the previous step, hence you'd may skip the pre_step in case the previous step was a redirect. Please, double check whether this is intended behavior.

Reply:
I just checked and it looks like the binary event is not updated when the pre-step hooks are called (it has the event from the previous step), so I have removed this check from the hooks.